src/App.js

```diff
@@ -25,44 +24,98 @@ const generateSquares = () => {
   return squares;
 };
 
+// Creating 2D array
+const reshapSquare = (arr) => {
+  const newArr = [];
+  while (arr.length) newArr.push(arr.splice(0, 3));
+  return newArr;
+};
+
 const App = () => {
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [nextPlayer, setNextPlayer] = useState(true);
+  const [gameWinner, setWinner] = useState(null);
+
+  const currentPlayer = nextPlayer ? PLAYER_1 : PLAYER_2;
+
+  const changeSquare = (id) => {
+    if (gameWinner !== null) {
+      return;
+    }
+
+    let flattenArray = squares.flat();
+    changedSquares = flattenArray.map((square) => {
```

Comment:
Interesting choice to flatten and then reshape! It works and is clear. You may also consider how you can access and change the square with the correct id in the 2D array.